SCClassLibrary/Common/Math/Rational.sc

```diff
@@ -0,0 +1,194 @@
+				this.denominator * aNumber.denominator)
+		} {
+			^aNumber.performBinaryOpOnRational('+', this, adverb)
+		};
+	}
+	
+	- { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				(this.numerator * aNumber.denominator) - (this.denominator * aNumber.numerator),
+				this.denominator * aNumber.denominator )
+		}  {
+			^aNumber.performBinaryOpOnRational('-', this, adverb)
+		};	
+	}
+	
+	* { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				this.numerator * aNumber.numerator,
+				this.denominator * aNumber.denominator)
+		} {
+			^aNumber.performBinaryOpOnRational('*', this, adverb)
+		};
+	}
+	
+	
+	/ { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				this.numerator * aNumber.denominator,
+				this.denominator * aNumber.numerator )
+		} {
+			^aNumber.performBinaryOpOnRational('/', this, adverb)
+		}
+	}
+	
+	== { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^((this.numerator * aNumber.denominator) == (this.denominator * aNumber.numerator))
+		} {
+			^aNumber.performBinaryOpOnRational('==', this, adverb)
+		}	
+	}
+	
+	!= { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^((this.numerator * aNumber.denominator) != (this.denominator * aNumber.numerator))
+		} {
+			^aNumber.performBinaryOpOnRational('!=', this, adverb)
+		}	
+	}
+	
+	pow { arg aNumber;
+		case
+		{ aNumber == 0 } { ^(1.as(this.species)) }
+		{ aNumber > 0  } { ^(this.species.new(this.numerator.pow(aNumber.asFloat), this.denominator.pow(aNumber.asFloat))) }
+		{ aNumber < 0  } {
+			if((this.numerator == 0).not) {
+				^this.reciprocal.pow(aNumber.abs)
+			} {
+				"Numerator can't be 0".warn;
+			};
+		};
+	}
+	
+	simplify { arg maxDenominator=20;
+		^this.asFloat.asRational(maxDenominator)
+	}
+	
+	< { arg aNumber;
+		^(this.asFloat < aNumber.asFloat)
+	}
+	
+	> { arg aNumber;
+		^(this.asFloat > aNumber.asFloat)
+	}
+	
+	<= { arg aNumber;
+		^(this.asFloat <= aNumber.asFloat)
+	}
+	
+	>= { arg aNumber;
+		^(this.asFloat >= aNumber.asFloat)
+	}
+	
+	isNegative {
+		^this.numerator.isNegative
+	}
+	
+	isPositive {
+		^this.numerator.isPositive
+	}
+	
+	neg {
+		^(this * (-1))
+	}
+	
+	mod {
+		if (this.isNegative) {
+			^this.neg
+		} {
+			^this
+		};
+	}
+	
+	max { arg aNumber;
+		var rationalArray, floatArray;
+		rationalArray = [this, aNumber];
+		floatArray = rationalArray.asFloat;
+		^rationalArray.at(floatArray.indexOf(floatArray.max));
+	}
+	
+	min { arg aNumber;
+		var rationalArray, floatArray;
+		rationalArray = [this, aNumber];
+		floatArray = rationalArray.asFloat;
+		^rationalArray.at(floatArray.indexOf(floatArray.min));
+	}
+	
+	sqrt { ^this.pow(this.species.new(1,2)) }
+	squared { ^this.pow(this.species.new(2,1)) }
+	cubed { ^this.pow(this.species.new(3,1)) }	
+	abs { ^this.species.new(this.numerator.abs.asInteger, this.denominator.abs.asInteger) }
+}
```

Comment:
eventually, `Rational` should understand all those messages that `SimpleNumber` does, apart from those which make no sense mathematically.

But at least `div` is missing, see Number: `div { arg aNumber; ^this.subclassResponsibility(thisMethod) }`